storage/indexer/src/indexer_reader.rs

```diff
@@ -17,6 +17,7 @@ use aptos_types::{
 };
 use std::sync::Arc;
 
+#[derive(Clone)]
 pub struct IndexerReaders {
     table_info_reader: Option<Arc<IndexerAsyncV2>>,
     db_indexer_reader: Option<Arc<DBIndexer>>,
```

Comment:
In all APIs, we should ensure the indexer's progress is over the requested version, otherwise we risk returning stale data.